test/app.js

```diff
@@ -188,7 +234,12 @@ describe('Smoke test', () => {
       done();
     };
 
-    const app = createTestAppWithNetworkCB(null, networkCb);
```

Comment:
`createTestAppWithNetworkCB` is replaced because in this PR it asynchronous. Since this test relies on `done`, if I try to turn it into an `async` test, an error is telling me to choose either to return a `promise` or `done`, but not both.

Reply:
I see, well then you can just use Promises then

Reply:
I think utilizing `done` is the only way since this test depends on callback. Returning a Promise, in this case, creates a false positive test.